packages/gafl-webapp-service/src/pages/order-complete/pdf-route.js

```diff
@@ -0,0 +1,43 @@
+import path from 'path'
+import Dirname from '../../../dirname.cjs'
+import { ORDER_COMPLETE_PDF } from '../../uri.js'
+import { PDF_FILENAME } from '../../constants.js'
+import { orderConfirmationPdf } from '../../processors/confirmation-pdf.js'
+import Pdfmake from 'pdfmake'
+
+const createPdfBinary = async pdfDoc => {
+  return new Promise(resolve => {
+    const printer = new Pdfmake({
+      Roboto: {
+        normal: path.join(Dirname, 'src/pages/fonts/Roboto-Regular.ttf'),
+        bold: path.join(Dirname, 'src/pages/fonts/Roboto-Medium.ttf'),
+        italics: path.join(Dirname, 'src/pages/fonts/Roboto-Italic.ttf'),
+        bolditalics: path.join(Dirname, 'src/pages/fonts/Roboto-MediumItalic.ttf')
+      }
+    })
+
+    const doc = printer.createPdfKitDocument(pdfDoc)
+    const chunks = []
+
+    doc.on('data', function (chunk) {
+      chunks.push(chunk)
+    })
+    doc.on('end', function () {
+      resolve(Buffer.concat(chunks))
+    })
+    doc.end()
```

Comment:
Should it be listening for the error event here and rejecting the promise?  As in:
```
doc.on('error', (err) => {
  reject(err);
});
```

Reply:
Error is caught later by the server exception handler - this is harder to mock and is not in the example code provided.